lib/edamam_api_wrapper.rb

```diff
@@ -0,0 +1,66 @@
+require 'httparty'
+
+class EdamamApiWrapper
+  BASE_URL = "https://api.edamam.com/search"
+  ID = ENV["APP_ID"]
+  KEY = ENV["APP_KEY"]
+
+  def self.search_recipes(q)
+    url =  BASE_URL + "?app_id=#{ID}" + "&app_key=#{KEY}" + "&q=#{q}"
+
+    encoded_url = URI.encode(url)
+    response = HTTParty.get(encoded_url)
+
+    recipe_list = []
+
+    if response["hits"]
+      response["hits"].each do |hit|
+        recipe_list << create_recipe(hit["recipe"])
+
+      end
+    end
+    return recipe_list
+  end
+
+  # this is to create a new instance of recipe when a show recipe is called
+  def self.show_recipe(id)
+    # this is like the encoding thing but for this show page
+
+    new_url = CGI.escape(uri_from_id(id))
+
+    # this is passing in the new url instead of a quiery
+    url = BASE_URL + "?app_id=#{ID}" + "&app_key=#{KEY}" + "&r=#{new_url}"
+
+    response = HTTParty.get(url)
+    # call on the first of the array of hashes
+    return create_recipe(response[0])
+  end
+
+  private
+
+  # using regex to get the last part of the uri
+  def self.id_from_uri(uri)
```

Comment:
I like that you broke all these out as separate methods and made them private - good organization.